config/v1/types_cmo.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=clustermonitoring,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1929
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:internal
+// ClusterMonitoring is the Schema for the clustermonitoringoperators API
+type ClusterMonitoring struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ClusterMonitoringSpec `json:"spec,omitempty"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status ClusterMonitoringStatus `json:"status,omitempty"`
+}
+
+// ClusterMonitoringOperatorStatus defines the observed state of ClusterMonitoringOperator
+type ClusterMonitoringStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:internal
+type ClusterMonitoringList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ClusterMonitoring `json:"items"`
+}
+
+// ClusterMonitoringOperatorSpec defines the desired state of ClusterMonitoringOperator
+type ClusterMonitoringSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// `UserWorkloadEnabled` is a Boolean flag that enables monitoring for user-defined projects.
+	UserWorkloadEnabled *bool `json:"enableUserWorkload,omitempty"`
```

Comment:
verify is failing due to:
```
F0612 18:28:37.679340 3391636 root.go:64] Error running codegen:                                                                                                
error running generator schemacheck on group config.openshift.io:                                                                                               
        could not run schemacheck generator for group/version config.openshift.io/v1:                                                                           
                error in config/v1/zz_generated.crd-manifests/clustermonitoring.crd.yaml: NoBools: crd/clustermonitoring.config.openshift.io version/v1 field/^.
spec.enableUserWorkload may not be a boolean                                                                                                                    
                error in config/v1/zz_generated.featuregated-crd-manifests/clustermonitoring.config.openshift.io/AAA_ungated.yaml: NoBools: crd/clustermonitorin
g.config.openshift.io version/v1 field/^.spec.enableUserWorkload may not be a boolean                                                                           
+ echo This verifier checks all files that have changed. In some cases you may have changed or renamed a file that already contained api violations, but you are
 not introducing a new violation. In such cases it is appropriate to /override the failing CI job.                                                              
This verifier checks all files that have changed. In some cases you may have changed or renamed a file that already contained api violations, but you are not in
troducing a new violation. In such cases it is appropriate to /override the failing CI job.                                                                     
make: *** [Makefile:65: verify-crd-schema] Error 255        
```                               

https://github.com/openshift/api/blob/5c2b46e4709a950e98cda1f3733f9a46cd9737cd/tools/vendor/github.com/openshift/crd-schema-checker/pkg/manifestcomparators/comp_no_bools.go#L20

how can we fix it?

Reply:
nevermind, add a enum as in https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#do-not-use-boolean-fields